### DISCUSSION TITLE
CurrentCollisionPairs Not Working As Intended

### DISCUSSION THREAD

**Discussion by jasoon-chen** [empriselab/RCareWorld#125] [Help - Programming] [answered]
I'm trying to retrieve the total collision pairs within the simulator. Based on the documentation provided [here](https://rcareworld.readthedocs.io/en/latest/api/pyrcareworld.envs.html#pyrcareworld.envs.base_env.RCareWorld.GetCurrentCollisionPairs) it mentions that "After calling this method and stepping once, the result will be saved in `env.data[‘CurrentCollisionPairs’]`." After running:
`print(env.data)` this is what is printed:

```{'time_scale': 1.0, 'fixed_delta_time': 0.019999999552965164, 'current_collisio_pairs': []}```

Despite running `print(env.data)` whenever there is a collision or not, the `current_collisio_pairs` key would always have an empty list `[]`. (I am assuming that the `CurrentCollisionPairs`key is basically `current_collisio_pairs`). Is this how we're supposed to grab the collision pairs or is there another function that we can use? 

This is the code that I was using:

```
self.env.GetCurrentCollisionPairs()
self.env.step()
print(self.env.data["current_collisio_pairs"])```

**Comment by YoruCathy** [comment]
Hi @jasoon-chen , we didn't enable this API in the dressing/bathing environment. To understand how the collision pair works, you can add your code to example_deug.py after this line https://github.com/empriselab/RCareWorld/blob/0eb1ad6c20d281fb386689819f0edefdf153417a/pyrcareworld/pyrcareworld/demo/examples/example_debug.py#L57
We can enable this function if this is a bottleneck for your algorithm

**Reply by jasoon-chen** [reply]
It would be much appreciated if this function could be enabled. Thanks.

**Reply by EltonCN** [reply]
Why can't we have a parallel version of the environment for training purposes, and the official with blocked functions only for scoring? In a real world problem, we would try to create a simulator to carry out the training, and then carry out the evaluation in the real environment.

So far my team has frequently had ideas that are made impossible by simulation, but which we know it supports from the simulator examples (which were made available by the simulator and is where we got the ideas from), and for undocumented reasons do not work in the competition environment.

**Reply by jasoon-chen** [reply]
Is it possible to get a follow up on this?

**Reply by DanielTakeshi** [reply]
@YoruCathy just wondering if it's possible to check this, thanks!

**Reply by YoruCathy** [reply]
Hi @DanielTakeshi we'll look into this issue soon.

**Reply by thisjustin123** [reply]
Hello! To clarify, this is a bug--not intentionally disabled--so I just made an issue for this. I'll look into this tonight and see if I can find out why this is happening.

EDIT: Found a fix, but have to run some things by Ruolin first before I can push:

![image](https://github.com/user-attachments/assets/e7f87672-e8a9-49d0-bbd8-3a982177ed02)

EDIT 2: #131

**Reply by YoruCathy** [reply, marked as answer]
Fixed in https://github.com/empriselab/RCareWorld/pull/131
